internal/metrics/v2/counter_test.go

```diff
@@ -0,0 +1,198 @@
+package v2
+
+import (
+	"testing"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNewCounter(t *testing.T) {
+	tests := []struct {
+		name        string
+		opts        CounterOpts
+		shouldPanic bool
+	}{
+		{
+			name: "should successfully create a counter",
+			opts: CounterOpts{
+				Subsystem:  "test_subsystem",
+				Name:       "counter_test_total",
+				Help:       "counter_test help",
+				LabelNames: []string{"ws_close_code"},
+			},
+		},
+		{
+			name:        "should panic when CounterOpts is empty",
+			opts:        CounterOpts{},
+			shouldPanic: true,
+		},
+	}
+	for _, test := range tests {
+		r := prometheus.NewRegistry()
+		t.Run(test.name, func(t *testing.T) {
+			if test.shouldPanic {
+				require.Panics(t, func() {
+					NewCounter(r, test.opts)
+				})
+			} else {
+				require.IsType(t, &prometheusCounter{}, NewCounter(r, test.opts))
+			}
+		})
+	}
+}
+
+func TestPrometheusCounterAdd(t *testing.T) {
+	type fields struct {
+		registry *prometheus.Registry
+		opts     CounterOpts
+	}
+	type args struct {
+		floatVal float64
+		label    []Label
+	}
+	tests := []struct {
+		name        string
+		fields      fields
+		args        args
+		expect      float64
+		shouldPanic bool
+	}{
+		{
+			name: "should add",
+			fields: fields{
+				registry: prometheus.NewRegistry(),
```

Comment:
nit: In these tests (not just this file, but all of 'em), we can just default the registry in for loop below.

Reply:
Agree. I'm going to leave it since its a nit. I originally planned to test behavior when using multiple registries but I don't think that's necessary at the moment.

Reply:
Yeah I don't think it's necessary anyway. Either way works with me.